src/sisl/sparse_geometry.py

```diff
@@ -3006,6 +3006,108 @@ def assert_unique(old, new):
         out._csr = csr
         out._geometry = geom
         return out
+    
+    def translate2uc(self):
```

Comment:
Probably the interface should be equivalent to the `Geometry.translate2uc`, it would also allow selection of `atoms`.

Reply:
Yes :+1: This was just to make it as simple as possible for now so that the code was easier to review